ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -466,6 +509,8 @@ private void initialize(final KsqlConfig configWithApplicationServer) {
         ksqlConfigNoPort
     );
     commandRunner.processPriorCommands();
+    cleanupOldStateDirectories(configWithApplicationServer);
```

Comment:
If we indeed have large amount of dangling state store dirs here there's a small risk that the restoring of existing queries -- i.e. starting the execution of them -- may already failed. Hence I'm wondering if we can refactor the timing of triggering this function a bit, i.e. after we've consumed through all the commands from the cmd topic, but before we start executing the restoration of queries?

Reply:
That makes sense to me, probably right here: https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java#L286. WDYT? I can make that shift

Reply:
Yeah that seems a better ordering to me.